@Ishiihara The jenkins failures are due to

```
/home/jenkins/jenkins-slave/workspace/kafka-pr-jdk10-scala2.12/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java:212: error: unreported exception RocksDBException; must be caught or declared to be thrown
18:24:17                     batch.remove(record.key);
18:24:17                                 ^
18:24:17 /home/jenkins/jenkins-slave/workspace/kafka-pr-jdk10-scala2.12/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java:214: error: unreported exception RocksDBException; must be caught or declared to be thrown
18:24:17                     batch.put(record.key, record.value);
```

--- a/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java
+++ b/streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java
@@ -207,11 +207,15 @@ class RocksDBSegmentedBytesStore implements SegmentedBytesStore {
                     // the store is already open here.
                     bulkLoadSegments = new HashSet<>(segments.allSegments());
                 }
-                final WriteBatch batch = writeBatchMap.computeIfAbsent(segment, s -> new WriteBatch());
-                if (record.value == null) {
-                    batch.remove(record.key);
-                } else {
-                    batch.put(record.key, record.value);
+                try {
+                    final WriteBatch batch = writeBatchMap.computeIfAbsent(segment, s -> new WriteBatch());
+                    if (record.value == null) {
+                        batch.remove(record.key);
+                    } else {
+                        batch.put(record.key, record.value);
+                    }
+                } catch (final RocksDBException e) {
+                    throw new ProcessorStateException("Error restoring batch to store " + this.name, e);
                 }
             }
         }

